R/init.R

```diff
@@ -161,8 +161,7 @@ init <- function(data,
 
   checkmate::assert_string(title, null.ok = TRUE)
   checkmate::assert_class(data, "TealData")
-  checkmate::check_list(modules)
-  checkmate::check_class(modules, "teal_modules")
+  checkmate::assert_multi_class(modules, c("list", "teal_modules"))
```

Comment:
So I think we should do this rather than allow a list:

```
checkmate::assert_multi_class(modules, c("teal_modules", "teal_modules"))
if (inherits(modules, "teal_modules") modules <- modules(modules)
```

We should probably update an example or two (e.g. the one for example_modules() ) to not use `modules()` and also update NEWS

Reply:
This would be a new feature and I tried to fix the bug only but in general you are right - that would be a nice enhancement. Let me do that.